library/StandardLibrary/src/Juvix/Library/Feedback.hs

```diff
@@ -0,0 +1,61 @@
+module Juvix.Library.Feedback where
+
+import qualified Control.Monad.Trans as Trans
+import Juvix.Library
+
+-- | Keep track of messages during the compilation or REPL.
+-- The implementation is based on
+-- [[https://github.com/UU-ComputerScience/uu-cco/blob/master/uu-cco/src/CCO/Feedback.hs]].
+data Feedback msg a
+  = Success msg a -- Indicate success.
+  | Fail msg -- Indicate a failure.
```

Comment:
I don't think it's a good idea having same `msg` for both Success and Fail.

I would expect having a CompileError datatype that encapsulates all errors. E.g.
```
data PipelineError =  ParseError ... | TypecheckError ... | CompileError ... | ...
```